src/admin/includes/formDefinitions/callbacks.php

```diff
@@ -0,0 +1,24 @@
+<?php
+function processInsert(){
+  $localvars = localvars::getInstance();
+  $formInfo  = $_POST['MYSQL'];
+
+  $roomID = isset($formInfo['roomID']) ? $formInfo['roomID'] : NULL;
+
+  $dupCheck  = reservationPermissions::duplicatePermissionsCheck($formInfo['resourceID'], $formInfo['resourceType'], $roomID, $formInfo['email']);
+
+  if (!$dupCheck) {
+   // return formInfo this is not a Duplicate Entry
```

Comment:
It would be bbetter to do something like (this is pseudo code)

try {
if (!$dupeCheck = reservationPermissions::duplicatePermissionsCheck))  {
throw new exceoption;
}
catch {
errorHandle::newError(__FUNCTION__."() - Error getting room name.", errorHandle::DEBUG); // debug can be replaced with the severity, including info
errorHandle::errorMsg("Duplicate Entry");
}